src/types.ts

```diff
@@ -37,6 +44,18 @@ export type FailureEventPayload<P extends EventPayload> = {
   };
 };
 
+export type FailureEventArgs<P extends EventPayload = EventPayload> = {
+  /**
+   * The event data present in the payload.
+   */
+  event: FailureEventPayload<P>;
+
+  /**
+   * The final error that caused this function to exhaust all retries.
+   */
+  err: FailureEventPayload<P>["data"]["error"];
```

Comment:
Why `err` and not `error`?

Reply:
@djfarrelly Errors are most commonly assigned as `err` in JS which I see. Do you think it's more reasonable to use the full version?

Reply:
I think `error` is a bit more explicit and clear. `err` feels like a convention as does `catch(e)`, but we write our `event` and `step` arguments in full - shortening `error` to `err` feels like a weird divergence. Unfortunately, with an options arg, one cannot rename this as they want, unless we change the error handler to be `onFailure(error, { event })`.

Reply:
@djfarrelly Great points. Cool, let's stick with `error` then - you're right that it's more consistent.